lib/puma/server.rb

```diff
@@ -386,7 +389,8 @@ def handle_servers
                 break if handle_check
               else
                 begin
-                  if io = sock.accept_nonblock
+                  receive_socket = sock.is_a?(UNIXSocket) && @update_capacity
+                  if (io = receive_socket ? sock.recv_io(TCPSocket) : sock.accept_nonblock)
```

Comment:
While this will work because sock will be a `UNIXServer` in the case of binding on a unix domain socket as the user requested, it's a bit clunky. I'd rather we plumb through a flag to indicated that `recv_io` should be used rather than relying on it being a UNIXSocket.